src/scalar/dusk.rs

```diff
@@ -253,6 +254,35 @@ impl Scalar {
         res
     }
 
+    /// Compute a uniformly distributed random scalar.
+    ///
+    /// Because scalars take 255 bits for encoding it is difficult to generate
+    /// random bit-pattern that ensures to encodes a valid scalar.
+    /// Wrapping the values that are higher than [`MODULUS`] results in hitting
+    /// some values more than others, and zeroing out the highest two bits will
+    /// eliminate some values from the possible results.
+    ///
+    /// This function achieves a uniform distribution of scalars by using
+    /// rejection sampling: random bit-patterns are generated until a valid
+    /// scalar is found.
+    /// The function is not constant time but that shouldn't be a concern since
+    /// no information about the scalar can be gained by knowing the time of
+    /// its generation.
+    pub fn uni_random(rng: &mut (impl RngCore + CryptoRng)) -> Self {
+        let mut buf = [0; 32];
+        let mut scalar: Option<Self> = None;
+
+        // we loop as long as it takes to generate a valid scalar
+        while scalar == None {
+            // Since modulus has 255-bits or less, we can zero the first bit
+            // to improve our chances of generating a valid scalar
+            buf[32 - 1] &= 0b0111_1111;
```

Comment:
Is this the first bit of the last byte right? If it's the expected behavior you should edit the comment above

Reply:
Furthermore, should you zeroize it after the random fill?

Reply:
totally!